sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -223,6 +223,13 @@ class SparkSession private(
   }
 
   /**
+   * Creates a new [[Dataset]] containing zero elements.
+   * @return 2.0.0
+   */
+  @Experimental
+  def emptyDataset[T: Encoder]: Dataset[T] = emptyDataFrame.as[T]
```

Comment:
cc @cloud-fan any idea how to properly implement this? The current implementation will crash.